src/p_pspr.c

```diff
@@ -1208,6 +1212,97 @@ static void P_NuggetBobbing(player_t* player)
   }
 }
 
+#define MAX_DELTA (ORIGWIDTH << FRACBITS)
+
+static void WeaponInertiaHorizontal(player_t* player, pspdef_t *psp)
+{
+  angle_t angle;
+  boolean clockwise;
+
+  if (player->mo->oldangle > player->mo->angle)
+  {
+    angle = player->mo->oldangle - player->mo->angle;
+    clockwise = (angle <= ANG180);
+    if (!clockwise)
+      angle = ANGLE_MAX - angle;
+  }
+  else
+  {
+    angle = player->mo->angle - player->mo->oldangle;
+    clockwise = (angle > ANG180);
+    if (clockwise)
+      angle = ANGLE_MAX - angle;
+  }
+
+  angle >>= ANGLETOFINESHIFT;
+  if (angle > 0)
+  {
+    const fixed_t scale = FRACUNIT - FixedDiv(angle << FRACBITS, FINEANGLES << FRACBITS);
+    const fixed_t delta = MAX_DELTA - FixedMul(MAX_DELTA, FixedMul(scale, scale));
+    psp->wix += clockwise ? -delta : delta;
+  }
+
+  if (psp->wix != 0)
+  {
+    psp->wix >>= 1;
```

Comment:
This and `psp->wiy >>= 1` a few lines down control how quickly the weapon returns to center.